src/LearnositySdk/Utils/Json.php

```diff
@@ -104,4 +110,63 @@ public static function isJson($val)
     {
         return !is_null(json_decode($val, true));
     }
+
+    /**
+     * This function builds up a map of float values in scientific notation and the equivalent float
+     * notations. json_encode encodes float values in scientific notation, but we want to make sure
+     * they are in float notation
+     *
+     * @param $json the $json to be encoded
+     * @param $floatMap floatMap in the current recursive call
+     * @return array mapping scientific values to float values
+     */
+    private static function getHighPrecisionFloatMap($value, $floatMap)
+    {
+        if (is_array($value)) {
+            // Recurse
+            foreach ($value as $entry) {
+                $floatMap = array_merge(static::getHighPrecisionFloatMap($entry, $floatMap), $floatMap);
+            }
+        } else {
+            // Terminate
+            $floatMap = array_merge(static::getFloatMap($value), $floatMap);
+        }
+        return $floatMap;
+    }
```

Comment:
I think we can simplify this method down to (taking into account the above comments):

```php
if (!is_array($value)) {
    return stataic::getFloatMap($value);
}

$floatMap = [];
foreach ($value as $entry) {
    $floatMap = array_merge(static::getHighPrecisionFloatMap($entry);
}
return $floatMap;
```